### ISSUE TITLE
DDAssert for Swift

### ISSUE THREAD

**Issue by ole** [CocoaLumberjack/CocoaLumberjack#918]
### New Issue Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/CocoaLumberjack/CocoaLumberjack/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](https://github.com/CocoaLumberjack/CocoaLumberjack#documentation) or [CocoaDocs](cocoadocs.org/docsets/CocoaLumberjack)
* [x] I have searched for a similar issue in the [project](https://github.com/CocoaLumberjack/CocoaLumberjack/issues) and found none

### Issue Description and Steps

It seems there is currently no equivalent for `DDAssert` in CocoaLumberjack's Swift API. Would you be open to adding this? I thought I'd ask before I send a pull request.

I'd like to see two Swift functions, mimicking what's available in the Swift stdlib: [`DDAssert`](https://developer.apple.com/documentation/swift/1541112-assert) and [`DDAssertionFailure`](https://developer.apple.com/documentation/swift/1539616-assertionfailure).

I imagine the implementation would look something like this:

```swift
public func DDAssert(_ condition: @autoclosure () -> Bool, _ message: @autoclosure () -> String = "", level: DDLogLevel = defaultDebugLevel, context: Int = 0, file: StaticString = #file, function: StaticString = #function, line: UInt = #line, tag: Any? = nil, asynchronous async: Bool = false, ddlog: DDLog = DDLog.sharedInstance) {
  if condition() == false {
    DDLogError(message, level: level, context: context, file: file, function: function, line: line, tag: tag, asynchronous: async, ddlog: ddlog)
    Swift.assertionFailure(message, file: file, line: line)
  }
}

public func DDAssertionFailure(_ message: @autoclosure () -> String = "", level: DDLogLevel = defaultDebugLevel, context: Int = 0, file: StaticString = #file, function: StaticString = #function, line: UInt = #line, tag: Any? = nil, asynchronous async: Bool = false, ddlog: DDLog = DDLog.sharedInstance) {
  DDLogError(message, level: level, context: context, file: file, function: function, line: line, tag: tag, asynchronous: async, ddlog: ddlog)
  Swift.assertionFailure(message, file: file, line: line)
}
```

_Edit: fixed the condition check in `DDAssert` (because I'm stupid) and added a default value for the `message` parameter so that you can call these without a message string._

**Comment by diederich** [comment]
+1 for this (as it just came up again :-)) 

@bpoplauschi thoughts on reviewing/merging if someone puts up a PR?

**Comment by bpoplauschi** [comment]
@ole @diederich I like this idea. If someone has the time to create a PR, I will make sure it is merged and published soon

**Comment by ole** [comment]
@bpoplauschi Cool. I'll send a PR in the next few days.

**Comment by ole** [comment]
Implemented in #934.
